gravitee-apim-console-webui/src/portal/customization/theme/portal-theme.component.ts

```diff
@@ -14,32 +14,214 @@
+  tertiaryColor?: string;
+  errorColor?: string;
+  pageBackgroundColor?: string;
+  cardBackgroundColor?: string;
+  customCSS?: string;
+}
 
 @Component({
   selector: 'portal-theme',
   templateUrl: './portal-theme.component.html',
   styleUrls: ['./portal-theme.component.scss'],
-  imports: [CommonModule, GioSaveBarModule, MatCardModule, MatFormFieldModule, MatInputModule, ReactiveFormsModule, PortalHeaderComponent],
+  imports: [
+    CommonModule,
+    GioFormFilePickerModule,
+    GioSaveBarModule,
+    MatCardModule,
+    MatFormFieldModule,
+    MatIconModule,
+    MatInputModule,
+    ReactiveFormsModule,
+    MatExpansionModule,
+    GioFormColorInputModule,
+    GioAvatarModule,
+    GioFormFilePickerModule,
+    MatAnchor,
+    MatButton,
+    MatTooltipModule,
+    GioMonacoEditorModule,
+    PortalHeaderComponent,
+  ],
   standalone: true,
 })
-export class PortalThemeComponent {
-  public developerPortalThemeForm = new FormGroup({
-    name: new FormControl(null, Validators.required),
-  });
+export class PortalThemeComponent implements OnInit, OnDestroy {
+  constructor(
+    private readonly uiPortalThemeService: UiPortalThemeService,
+    private readonly snackBarService: SnackBarService,
+  ) {}
+
+  private initialTheme: ThemePortalNext;
+  private defaultValues: ThemeVM;
+  private initialValues: ThemeVM;
+
+  public portalThemeForm;
+
+  private unsubscribe$: Subject<void> = new Subject<void>();
+
+  formUnchanged: boolean = true;
+
+  ngOnInit() {
+    this.uiPortalThemeService
+      .getDefaultTheme('PORTAL_NEXT')
+      .pipe(
+        tap((theme: ThemePortalNext) => {
+          this.defaultValues = this.convertThemeToThemeVM(theme);
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe();
+
+    this.uiPortalThemeService
+      .getCurrentTheme('PORTAL_NEXT')
+      .pipe(
+        tap((theme: ThemePortalNext) => {
+          this.initialTheme = theme;
+        }),
+        map((theme: ThemePortalNext) => this.convertThemeToThemeVM(theme)),
+        tap((themeVM: ThemeVM) => {
+          this.initialValues = themeVM;
+          this.portalThemeForm = new FormGroup({
+            logo: new FormControl(this.initialValues.logo),
+            favicon: new FormControl(this.initialValues.favicon),
+            font: new FormControl(this.initialValues.font),
+            primaryColor: new FormControl(this.initialValues.primaryColor),
+            secondaryColor: new FormControl(this.initialValues.secondaryColor),
+            tertiaryColor: new FormControl(this.initialValues.tertiaryColor),
+            errorColor: new FormControl(this.initialValues.errorColor),
+            pageBackgroundColor: new FormControl(this.initialValues.pageBackgroundColor),
+            cardBackgroundColor: new FormControl(this.initialValues.cardBackgroundColor),
+            customCSS: new FormControl(this.initialValues.customCSS),
+          });
+
+          this.portalThemeForm.valueChanges
```

Comment:
Non-blocking comment: this can probably go outside of the `this.uiPortalThemeService` observable chain. Best practice is to not have a subscribe within a observable that has a subscribe at the end.